modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -371,6 +371,26 @@ def test_loc(data):
     with pytest.raises(KeyError):
         modin_df.loc["NO_EXIST"]
 
+    # From issue #3764
+    pandas_df = pandas.DataFrame([[1, 2, 3], [4, 5, 6]])
+    modin_df = pd.DataFrame([[1, 2, 3], [4, 5, 6]])
+
+    pandas_df.loc[2] = pandas_df.loc[1]
+    modin_df.loc[2] = modin_df.loc[1]
+    df_equals(modin_df, pandas_df)
+
+    pandas_df.loc[6] = pandas_df.loc[1]
+    modin_df.loc[6] = modin_df.loc[1]
+    df_equals(modin_df, pandas_df)
+
+    pandas_df.loc[lambda df: 70] = pandas_df.loc[1]
+    modin_df.loc[lambda df: 70] = modin_df.loc[1]
+    df_equals(modin_df, pandas_df)
+
+    pandas_df.loc[90] = pandas_df.loc[70]
+    modin_df.loc[90] = modin_df.loc[70]
```

Comment:
what about the same case, but when a column is appended?
```python
>>> df
   a  b  c
1  1  2  3
2  1  2  3
3  1  2  3
>>> df.loc[:, "d"] = 1
>>> df
   a  b  c  d
1  1  2  3  1
2  1  2  3  1
3  1  2  3  1
```
This PR only changes the [logic of inserting a new row](https://github.com/RehanSD/modin/blob/032007187ac34f3f99205da36063399600a4ac40/modin/pandas/indexing.py#L722-L725), but not the column. Isn't [column inserting](https://github.com/RehanSD/modin/blob/032007187ac34f3f99205da36063399600a4ac40/modin/pandas/indexing.py#L731-L735) is affected by the same bug that was fixed there for rows?

Reply:
I added tests for out of bounds columns in the most recent commit to cover this case.